worlds/tunic/logic_helpers.py

```diff
@@ -0,0 +1,98 @@
+from typing import TYPE_CHECKING
+
+from BaseClasses import CollectionState
+
+from .constants import *
+from .fuses import fuse_activation_reqs
+from .options import HexagonQuestAbilityUnlockType, IceGrappling
+
+if TYPE_CHECKING:
+    from . import TunicWorld
+
+
+def randomize_ability_unlocks(world: "TunicWorld") -> dict[str, int]:
+    options = world.options
+
+    abilities = [prayer, holy_cross, icebolt]
+    ability_requirement = [1, 1, 1]
+    world.random.shuffle(abilities)
+
+    if options.hexagon_quest.value and options.hexagon_quest_ability_type == HexagonQuestAbilityUnlockType.option_hexagons:
+        hexagon_goal = options.hexagon_goal.value
+        # Set ability unlocks to 25, 50, and 75% of goal amount
+        ability_requirement = [hexagon_goal // 4, hexagon_goal // 2, hexagon_goal * 3 // 4]
+        if any(req == 0 for req in ability_requirement):
+            ability_requirement = [1, 2, 3]
+
+    return dict(zip(abilities, ability_requirement))
+
+
+def has_ability(ability: str, state: CollectionState, world: "TunicWorld") -> bool:
+    options = world.options
+    ability_unlocks = world.ability_unlocks
+    if not options.ability_shuffling:
+        return True
+    if options.hexagon_quest and options.hexagon_quest_ability_type == HexagonQuestAbilityUnlockType.option_hexagons:
+        return state.has(gold_hexagon, world.player, ability_unlocks[ability])
+    return state.has(ability, world.player)
+
+
+# a check to see if you can whack things in melee at all
+def has_melee(state: CollectionState, player: int) -> bool:
+    return state.has_any({"Stick", "Sword", "Sword Upgrade"}, player)
```

Comment:
```suggestion
    return state.has_any(("Stick", "Sword", "Sword Upgrade"), player)
```
and similar for the rest of the file